src/main/user-impl/java/com/mysql/cj/jdbc/ha/plugins/failover/AuroraTopologyService.java

```diff
@@ -211,19 +210,24 @@ protected ClusterTopologyInfo queryForTopology(JdbcConnection conn) throws SQLEx
     long startTimeMs = System.currentTimeMillis();
 
     ClusterTopologyInfo topologyInfo = null;
-
-    try (Statement stmt = conn.createStatement()) {
-      try (ResultSet resultSet = stmt.executeQuery(RETRIEVE_TOPOLOGY_SQL)) {
-        topologyInfo = processQueryResults(resultSet);
-      }
-    } catch (SQLSyntaxErrorException e) {
-      // We may get SQLSyntaxErrorException like the following from MySQL databases:
-      // "Unknown table 'REPLICA_HOST_STATUS' in information_schema"
-      // Ignore this kind of exceptions.
-    } finally {
-      if (gatherPerfMetrics(conn.getPropertySet())) {
-        long currentTimeMs = System.currentTimeMillis();
-        this.metricsContainer.registerTopologyQueryExecutionTime(currentTimeMs - startTimeMs);
+    if (!conn.isInGlobalTx() && !conn.isInPreparedTx()) {
+      synchronized (this) {
```

Comment:
Why do we need synchronized here?

Reply:
I am concerned that the state of the transaction changes between when we check it and when we send out the query.